sched/group/group_create.c

```diff
@@ -138,6 +139,9 @@ int group_allocate(FAR struct task_tcb_s *tcb, uint8_t ttype)
       return -ENOMEM;
     }
 
+  group->tg_info = (FAR struct task_info_s *)
+    group_zalloc(group, sizeof(struct task_info_s));
```

Comment:
group_malloc looks at the GROUP_FLAG_PRIVILEGED bit, which has not been set yet here. is it intended?

is task_info_s planned to be used for GROUP_FLAG_PRIVILEGED at all?

i'm still not sure how this can work for CONFIG_BUILD_KERNEL. (i'm just asking here because i dunno.)

Reply:
> group_malloc looks at the GROUP_FLAG_PRIVILEGED bit, which has not been set yet here. is it intended?
> 

Yes, the code need move to:
https://github.com/apache/incubator-nuttx/pull/3992/files#diff-2b2403bb769be0921afc3e368b8f5e44acd58e6d8e97f2bc1e66064e3ca2de28L169

> is task_info_s planned to be used for GROUP_FLAG_PRIVILEGED at all?

The current kernel thread implementation is actually a normal task with privileged permission. task_info_s is a critical data structure associate with many key functions.

> 
> i'm still not sure how this can work for CONFIG_BUILD_KERNEL. (i'm just asking here because i dunno.)

The whole path is a follow up of: https://github.com/apache/incubator-nuttx/pull/997 and no body report that #997 break the kernel build. Since #997 enter the mainline more than two years, I think the approach should work in KERNEL mode.

Reply:
> > i'm still not sure how this can work for CONFIG_BUILD_KERNEL. (i'm just asking here because i dunno.)
> 
> The whole path is a follow up #997 and no body report that #997 break the kernel build. Since #997 enter the mainline more than two years, I think the approach should work in KERNEL mode.

i'm not saying it doesn't work. i just don't understand.
specifically, i don't know how addenv and kumm allocator interacts.
do you imply someone is actually running the kernel mode these days?

Reply:
> i'm not saying it doesn't work. i just don't understand.
> specifically, i don't know how addenv and kumm allocator interacts.

The rule is simple: switch to the target addrenv before calling any kumm_ or group_ API. After this, the heap allocation for kernel mode is almost same as the protected mode.

> do you imply someone is actually running the kernel mode these days?

I don't know, since we just use the flat and protected mode.

Reply:
ok. thank you for explanation.
after re-reading spawn and relevant code, this seems ok to me.